test/runner/e2e_read_test.cpp

```diff
@@ -45,6 +45,8 @@ void parseAndRegisterTestGroup(const std::string& path) {
                     return new EndToEndReadTest(dataset, bufferPoolSize, std::move(testStatements));
                 });
         }
+    } else {
+        throw Exception("Invalid test file");
```

Comment:
Can you add the file path into the exception here?
Also, I would add a `TestException` class and throw `TestException` here. See `common/exception.h`.